errors/errors.go

```diff
@@ -87,3 +90,50 @@ func (ce codedError) Is(err error) bool {
 const (
 	ErrUncoded Code = "Uncoded"
 )
+
+// MarshalJSON returns the provided error as a json object (as a string)
+// representing a codedError. If err is not already a codedError, the json
+// object will still represent a codedError but its `code` value will be empty.
+// Note: an empty code here is intentional and is different from code
+// `errors.Uncoded` which is a valid code; it just means the developer returned
+// a codedError but didn't bother to choose (or create) a useful error code.
+func MarshalJSON(err error) string {
```

Comment:
👍

Reply:
This is a good question. I have a few thoughts on how I imagined this evolving:

- add a test which verifies that the methods implemented by a service's http handler and corresponding client include this encoding (the goal is to prevent a future developer from adding a method which does not container the encoding)
- start logging (as a warning) any time this encoding path receives a non `codedError` so we can see what's getting missed
- over time, try to address those things being logged with the goal of eventually converting that log to a developer error (as you mentioned)
- go through the code and replace existing `UncodedError`s with an actual code.